### ISSUE TITLE
[Crash] error: Setting value twice for same attribute

### ISSUE THREAD

**Issue by sungwoncho** [dgraph-io/dgraph#998]
version: 0.7.7

---

*Reproduce*

1. Load the following data

[dgraph-1-2017-05-31-11-43.rdf.gz](https://github.com/dgraph-io/dgraph/files/1040257/dgraph-1-2017-05-31-11-43.rdf.gz)
[dgraph-schema-1-2017-05-31-11-43.rdf.gz](https://github.com/dgraph-io/dgraph/files/1040256/dgraph-schema-1-2017-05-31-11-43.rdf.gz)

2. Execute the following query.

```
{
    question(id: 0x757df9ecca177644) {
      Has.Answer {
        Body {
          Text
        }
        Owner {
          DisplayName
          Reputation
          _uid_
        }
        Timestamp
        UpvoteCount: count(Upvote)
        DownvoteCount: count(Downvote)
        History: ~Post(orderdesc: Timestamp) {
          
        Author {
          DisplayName
          Reputation
          _uid_
        }
        Type
        Text

          x as Timestamp
        }
        x1 as max(var(x))
      }
      x2 as max(var(x1))

      History: ~Post(orderdesc: Timestamp) {
        
      Author {
        DisplayName
        Reputation
        _uid_
      }
      Type
      Text

        y as Timestamp
      }
      y1 as max(var(y))

      lastActive as math(max(x2, y1))
      var(lastActive)
    }
  }
```

3. Dgraph crashes with the following log without stack trace.

```
2017/05/31 11:45:04 error.go:76: Setting value twice for same attribute var(lastActive)
```

**Comment by MichaelJCompton** [comment]
Just had the same error.  Seems to be whenever there is a math var that is used without an alias.  So
```
   lastActive as math(max(x2, y1))
   blah : var(lastActive)
```
would be fine.

Would it also be worth allowing alias on math as a shortcut on the above?  It's allowed on min, max, count etc. so is `blah : p as count(...)`.

**Comment by pawanrawal** [comment]
This is related to not being allowed to retrieve results for math expressions in the same block that they are defined. For math expressions, we have to define a variable because that's what we use to return the result, though we return an error when its not used again.

I think we should relax the requirement on variables so that they can be defined and not be used if its a math variable. This would allow retrieving of math values.

**Comment by pawanrawal** [comment]
Fix has been merged to `master` and `release/v0.7.7`.
